### PR TITLE
UI: Qt 6 deprecations

### PR DESCRIPTION
**Pull request by gxalpha**
### Description

Fixes deprecations for Qt.

### Motivation and Context

Less deprecations.

### How Has This Been Tested?

First commit:
- Undo/Redo hotkeys still work
- Moving items in the preview with arrow keys still works (with shift as well)

Second commit:
- Context Bar Media Slider (MediaSlider)
- Quick Transitions Button (MenuButton)
- Drag and Drop in Sources Dock and Scenes Dock (SourceTree and SceneTree)
- Interact Menu on Browser Source, with pointerpointer.com and webgl aquarium (OBSBasicInteraction)
- In preview, moving the mouse over items correctly highlights the item where the mouse is. When dragging a selection, that selection is in the correct place
- In multiview, clicking a scene correctly selects it for the preview. With "Transition to scene when double-clicked" enabled, double clicking a scene selects and transitions to it.

Third commit:
- Raised a sample information, warning, and critical message box
- Raised several question dialogs in OBS, like closing the properties window without saving or selecting lossless mode, and made sure the buttons have the intended effect.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/window-basic-preview.cpp [hunk 1]
```diff
@@ -499,11 +499,18 @@ void OBSBasicPreview::wheelEvent(QWheelEvent *event)
 
 void OBSBasicPreview::mousePressEvent(QMouseEvent *event)
 {
+
```
Comment (RytoEX) [inline UI/window-basic-preview.cpp:502]: Nit: can remove this blank line.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Did a quick visual review.

**Comment by tytan652** [comment]
Build against #6759 with Qt6 and got those lines as errors:
- https://github.com/obsproject/obs-studio/blob/master/UI/obs-app.cpp#L2127
- https://github.com/obsproject/obs-studio/blob/master/UI/obs-app.cpp#L2129
- https://github.com/obsproject/obs-studio/blob/master/UI/window-basic-main.cpp#L379 (Linux only)

Otherwise it builds fine.

**Comment by gxalpha** [comment]
@tytan652 should be fixed, feel free to re-test

**Comment by tytan652** [comment]
I confirm that this now that this PR as it is builds against #6759 with no errors without any fix required.

**Review by RytoEX — APPROVED** [review]
All my tests in Qt 6.3.0-git (close to 6.3.1) worked fine. I think @gxalpha had tested this in Qt5 for good measure, and CI is still building against as well, so at minimum we know it builds without warnings/errors.

**Comment by RytoEX** [comment]
Ran through the same tests I did yesterday, but with Qt 5.15.2. Everything seemed to work fine.

**Pull request merged**